src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -391,12 +391,19 @@ class AssemblerConnectionManager(
             entitySet: EntitySet,
             materializedPropertyTypes: Map<UUID, PropertyType>
     ): IntArray {
-
+        val permissionsToCheck = EdmAuthorizationHelper.READ_PERMISSION
         // collect all principals of type user, role, which have read access on entityset
         val authorizedPrincipals = securePrincipalsManager
-                .getAuthorizedPrincipalsOnSecurableObject(AclKey(entitySet.id), EdmAuthorizationHelper.READ_PERMISSION)
+                .getAuthorizedPrincipalsOnSecurableObject(AclKey(entitySet.id), permissionsToCheck)
                 .filter { it.type == PrincipalType.USER || it.type == PrincipalType.ROLE }
-                .toSet()
+                .filter {
+                    //There are some non-existing principals with permissions
+                    val principalExists = securePrincipalsManager.principalExists(it)
```

Comment:
This is a relatively heavy call since it executes a hazelcast predicate -- it might be easier to just try/catch on the lookup call that fails (or turn this into a bulk principalExists call)

Reply:
:+1: